### PR TITLE
27 visual improvements

### PR DESCRIPTION
**Pull request by PatrykSzwer**
#27

### CODE REVIEW

File: src/KitchenSink/wwwroot/KitchenSink/NavPage.html [hunk 1]
```diff
@@ -3,60 +3,55 @@
         <div slot="kitchensink/nav-navigation" class="kitchensink-layout">
```
Comment (warpech) [inline src/KitchenSink/wwwroot/KitchenSink/NavPage.html]: I don't think that this new grouping of pages is more understandable. I tried to think about the optimum groups, and this is what I got. WDYT?

#### Rendering

- Breadcrumb
- Datagrid
- HTML (renamed from Html)
- Link (renamed from Url)
- Map
- Markdown
- Pagination
- Table

#### Server Push

- Async Response (renamed from Callback behavior)
- Cookies (renamed from Set a Cookie)
- Dialog Box
- Flash Message
- Lazy Loading
- Nested Views (renamed from Nested partials)
- Progress Bar
- Redirect

#### User Input

- Autocomplete
- Button
- Checkbox
- Datepicker
- Dropdown
- Decimal Input (renamed from Decimal)
- File Upload
- Integer Input (renamed from Integer)
- Multiselect
- Password
- Radio Button (renamed from Radio)
- Radiolist
- Text Input (renamed from Text)
- Textarea
- Toggle Button (renamed from Toggle)
- Validation

File: src/KitchenSink/wwwroot/KitchenSink/CallbackPage.html [hunk 2]
```diff
@@ -1,12 +1,12 @@
 ﻿<template>
     <template is="dom-bind">
-        <h1 slot="kitchensink/callback-heading">Callback example</h1>
+        <h1 slot="kitchensink/callback-heading" class="kitchensink-heading-1">Callback behavior</h1>
```
Comment (warpech) [inline src/KitchenSink/wwwroot/KitchenSink/CallbackPage.html]: Change the header to "Async Response"

File: src/KitchenSink/wwwroot/KitchenSink/CookiePage.html [hunk 3]
```diff
@@ -1,6 +1,7 @@
 ﻿<template>
     <template is="dom-bind">
-        <h2 slot="kitchensink/cookie-heading">Setting cookies on server side</h2>
+        <h1 slot="kitchensink/cookie-heading" class="kitchensink-heading-1">Setting cookies on server side</h1>
```
Comment (warpech) [inline src/KitchenSink/wwwroot/KitchenSink/CookiePage.html]: Change the header to "Cookies" and add a first paragraph saying "This page sets a cookie when you load it."

The second paragraph can stay as it is.

File: src/KitchenSink/wwwroot/KitchenSink/FileUploadPage.html [hunk 4]
```diff
@@ -2,15 +2,15 @@
 
 <template>
     <template is="dom-bind">
-        <h1 slot="kitchensink/fileUpload-header">File upload example</h1>
-        
+        <h1 slot="kitchensink/fileUpload-heading" class="kitchensink-heading-1">File upload</h1>
```
Comment (warpech) [inline src/KitchenSink/wwwroot/KitchenSink/FileUploadPage.html]: Please change the `slot` attribute values to be always lowercase.

File: src/KitchenSink/wwwroot/KitchenSink/GeoPage.html [hunk 5]
```diff
@@ -3,7 +3,9 @@
 
 <template>
     <template is="dom-bind">
-        <p slot="kitchensink/geo-header" class="lead">JSON's primitive value type <code>number</code> can be used to provide latitude/longitude information for a map custom element, e.g. <a href="https://elements.polymer-project.org/elements/google-map">google-maps</a>.
+        <h1 slot="kitchensink/geo-heading" class="kitchensink-heading-1">Geo</h1>
```
Comment (warpech) [inline src/KitchenSink/wwwroot/KitchenSink/GeoPage.html]: Please change the heading to say "Map". 

Until https://github.com/StarcounterApps/KitchenSink/issues/10 is fixed, the filename and the URI of this page needs to remain "Geo".

File: src/KitchenSink/wwwroot/KitchenSink/TextPage.html [hunk 6]
```diff
@@ -2,29 +2,31 @@
 
 <template>
     <template is="dom-bind">
+        <h1 slot="kitchensink/text-heading" class="kitchensink-heading-1">Text</h1>
```
Comment (warpech) [inline src/KitchenSink/wwwroot/KitchenSink/TextPage.html]: Change the heading to "Text Input"

File: src/KitchenSink/wwwroot/KitchenSink/UrlPage.html [hunk 7]
```diff
@@ -1,12 +1,15 @@
 ﻿<template>
     <template is="dom-bind">
+        <h1 slot="kitchensink/url-heading" class="kitchensink-heading-1">URL</h1>
```
Comment (warpech) [inline src/KitchenSink/wwwroot/KitchenSink/UrlPage.html]: Change the heading to "Link"

File: src/KitchenSink/wwwroot/KitchenSink/DecimalPage.html [hunk 8]
```diff
@@ -1,11 +1,15 @@
 ﻿<template>
     <template is="dom-bind">
-        <p slot="kitchensink/decimal-heading" class="lead">JSON's primitive value type <code>number</code> can be mapped directly to HTML form elements <code>input type="number"</code>.</p>
+        <h1 slot="kitchensink/decimal-heading" class="kitchensink-heading-1">Decimal</h1>
```
Comment (warpech) [inline src/KitchenSink/wwwroot/KitchenSink/DecimalPage.html]: Change the heading to "Decimal Input"

File: src/KitchenSink/wwwroot/KitchenSink/IntegerPage.html [hunk 9]
```diff
@@ -1,8 +1,10 @@
 <template>
     <template is="dom-bind">
-        <p slot="kitchensink/integer-header" class="lead">JSON's primitive value type <code>number</code> can be mapped directly to HTML form elements <code>input type="number"</code>.</p>
+        <h1 slot="kitchensink/integer-heading" class="kitchensink-heading-1">Integer</h1>
```
Comment (warpech) [inline src/KitchenSink/wwwroot/KitchenSink/IntegerPage.html]: Change the heading to "Integer Input"

File: src/KitchenSink/wwwroot/KitchenSink/RadioPage.html [hunk 10]
```diff
@@ -3,6 +3,8 @@
 
 <template>
     <template is="dom-bind">
+        <h1 slot="kitchensink/radio-heading" class="kitchensink-heading-1">Radio</h1>
```
Comment (warpech) [inline src/KitchenSink/wwwroot/KitchenSink/RadioPage.html]: Change the heading to "Radio Button"

File: src/KitchenSink/wwwroot/KitchenSink/AutocompletePage.html [hunk 11]
```diff
@@ -1,6 +1,6 @@
 ﻿<template>
     <template is="dom-bind">
-        <h1 slot="kitchensink/autocomplete-heading">Autocomplete</h1>
+        <h1 slot="kitchensink/autocomplete-heading" class="kitchensink-heading-1">Autocomplete</h1>
 
         <p slot="kitchensink/autocomplete-products-description" class="lead">This autocomplete field uses DB as source. Enter <code>'*'</code> to show all available items</p>
```
Comment (warpech) [inline src/KitchenSink/wwwroot/KitchenSink/AutocompletePage.html]: I think that the `lead` classes look quite strange and unexpected. Maybe make all paragraphs look the same?

File: src/KitchenSink/wwwroot/KitchenSink/CallbackPage.html [hunk 12]
```diff
@@ -25,7 +25,7 @@ <h4 slot="kitchensink/callback-save-server-heading">Save button with server side
             </template>
         </div>
         <button slot="kitchensink/callback-save-server-button" type="button" class="btn btn-sm btn-default" value="{{model.SaveAndMessageTrigger$::click}}" disabled="{{model.SaveAndMessageTrigger$}}" onmouseup="++this.value;">Randomly save or fail changes (1s)</button>
-        <h4 slot="kitchensink/callback-save-client-heading">Save button with client side message</h4>
+        <h2 slot="kitchensink/callback-save-client-heading" class="kitchensink-heading-2">Save button with client side message</h2>
         <small slot="kitchensink/callback-save-client-note">
```
Comment (warpech) [inline src/KitchenSink/wwwroot/KitchenSink/CallbackPage.html]: I think that the small text looks strange. Maybe make all paragraphs formatted in the same way?

File: src/KitchenSink/wwwroot/KitchenSink/NavPage.html (hunk 1 above)
Comment (PatrykSzwer) [inline src/KitchenSink/wwwroot/KitchenSink/NavPage.html]: I think It's step into same direction as my changes but definitely much cleaner. Let's give it a shot!

File: src/KitchenSink/wwwroot/KitchenSink/DropdownPage.html [hunk 13]
```diff
@@ -2,9 +2,9 @@
 
 <template>
     <template is="dom-bind">
-        <h1 slot="kitchensink/dropdown-header">Primitive array</h1>
+        <h1 slot="kitchensink/dropdown-heading" class="kitchensink-heading-1">Dropdown</h1>
 
-        <p slot="kitchensink/dropdown-pets-description" class="lead">A combination of JSON's primitive value types <code>array</code>, <code>object</code> and <code>number</code> can be mapped to an HTML <code>select</code> element.</p>
+        <p slot="kitchensink/dropdown-pets-description">A combination of JSON's primitive value types <code>array</code>, <code>object</code> and <code>number</code> can be mapped to an HTML <code>select</code> element.</p>
```
Comment (warpech) [inline src/KitchenSink/wwwroot/KitchenSink/DropdownPage.html]: Please add a header "Primitive array" above this paragraph.

File: src/KitchenSink/wwwroot/KitchenSink/DropdownPage.html [hunk 14]
```diff
@@ -14,9 +14,9 @@ <h1 slot="kitchensink/dropdown-header">Primitive array</h1>
 
         <div slot="kitchensink/dropdown-pets-reaction" class="kitchensink-test-pet-like-label">{{model.PetReaction}}</div>
 
-        <p slot="kitchensink/dropdown-objects-header" class="lead">Objects array</p>
+        <p slot="kitchensink/dropdown-objects-header">Objects array</p>
```
Comment (warpech) [inline src/KitchenSink/wwwroot/KitchenSink/DropdownPage.html]: Please change this `p` to a header.

File: src/KitchenSink/wwwroot/KitchenSink/DropdownPage.html [hunk 15]
```diff
@@ -2,9 +2,9 @@
 
 <template>
     <template is="dom-bind">
-        <h1 slot="kitchensink/dropdown-header">Primitive array</h1>
+        <h1 slot="kitchensink/dropdown-heading" class="kitchensink-heading-1">Dropdown</h1>
 
-        <p slot="kitchensink/dropdown-pets-description" class="lead">A combination of JSON's primitive value types <code>array</code>, <code>object</code> and <code>number</code> can be mapped to an HTML <code>select</code> element.</p>
+        <p slot="kitchensink/dropdown-pets-description">A combination of JSON's primitive value types <code>array</code>, <code>object</code> and <code>number</code> can be mapped to an HTML <code>select</code> element.</p>
 
         <select slot="kitchensink/dropdown-pets-dropdown" class="kitchensink-select kitchensink-test-pets-select" value="{{model.SelectedPet$::change}}">
```
Comment (warpech) [inline src/KitchenSink/wwwroot/KitchenSink/DropdownPage.html]: This element now appears with wrong width.

Was:

2017-05-24 09_23_16-kitchensink



Is:

2017-05-24 09_22_54-kitchensink

File: src/KitchenSink/wwwroot/KitchenSink/NestedPartial.html [hunk 16]
```diff
@@ -1,6 +1,8 @@
 ﻿<template>
     <template is="dom-bind">
-        <p slot="kitchensink/nested-description" class="lead">JSON's primitive value type <code>object</code> can be used for nesting of the partials. This also works for recursive nesting.</p>
+        <h1 slot="kitchensink/nested-heading" class="kitchensink-heading-1">Nested partial</h1>
```
Comment (warpech) [inline src/KitchenSink/wwwroot/KitchenSink/NestedPartial.html]: Could you pls review all the `h1`s in every page? They should be the same as the entry in the `NavPage.html`

File: src/KitchenSink/wwwroot/KitchenSink/NavPage.html [hunk 17]
```diff
@@ -3,60 +3,49 @@
...
-                    <li><a href="/KitchenSink/Table">Table</a></li>
-                    <li><a href="/KitchenSink/Datagrid">Datagrid</a></li>
-                    <li><a href="/KitchenSink/Chart">Chart</a></li>
-                    <!--<li><a href="/KitchenSink/Navigation">Navigation</a></li> not ready yet-->
-                    <li><a href="/KitchenSink/Breadcrumb">Breadcrumb</a></li>
-                </ul>
-                <h3 class="kitchensink-menu__heading">Custom</h3>
-                <ul class="kitchensink-menu__list">
+                    <li><a href="/KitchenSink/Decimal">Decimal Input</a></li>
                     <li><a href="/KitchenSink/FileUpload">File upload</a></li>
```
Comment (warpech) [inline src/KitchenSink/wwwroot/KitchenSink/NavPage.html]: Should be "File Upload" (both words start from the capital letter)

File: src/KitchenSink/wwwroot/KitchenSink/NavPage.html [hunk 18]
```diff
@@ -3,60 +3,49 @@
...
-                <h3 class="kitchensink-menu__heading">Number</h3>
+                <h3 class="kitchensink-menu__heading"><b>Server Push</b> <span class="glyphicon glyphicon-transfer"></span></h3>
                 <ul class="kitchensink-menu__list">
-                    <li><a href="/KitchenSink/Integer">Integer</a></li>
-                    <li><a href="/KitchenSink/Decimal">Decimal</a></li>
-                    <li><a href="/KitchenSink/Button">Button</a></li>
-                    <li><a href="/KitchenSink/Geo">Map</a></li>
+                    <li><a href="/KitchenSink/Callback">Async Response</a></li>
+                    <li><a href="/KitchenSink/Cookie">Cookies</a></li>
+                    <li><a href="/KitchenSink/Dialog">Dialog box</a></li>
```
Comment (warpech) [inline src/KitchenSink/wwwroot/KitchenSink/NavPage.html]: Should be "Dialog Box" (both words start from the capital letter)

File: src/KitchenSink/wwwroot/KitchenSink/NavPage.html [hunk 19]
```diff
@@ -3,60 +3,49 @@
         <div slot="kitchensink/nav-navigation" class="kitchensink-layout">
             <nav class="kitchensink-layout__column-left">
                 <ul class="kitchensink-menu__list">
-                    <li><a href="/KitchenSink/MainPage">MainPage</a></li>
+                    <li><a href="/KitchenSink/MainPage"><span class="glyphicon glyphicon-home"></span></a></li>
```
Comment (warpech) [inline src/KitchenSink/wwwroot/KitchenSink/NavPage.html]: I still prefer the old formatting of the text in the `NavPage.html` - aligned to the left (not center) and without the icons.

File: src/KitchenSink/wwwroot/KitchenSink/GeoPage.html [hunk 20]
```diff
@@ -3,7 +3,9 @@
 
 <template>
     <template is="dom-bind">
-        <p slot="kitchensink/geo-header" class="lead">JSON's primitive value type <code>number</code> can be used to provide latitude/longitude information for a map custom element, e.g. <a href="https://elements.polymer-project.org/elements/google-map">google-maps</a>.
+        <h1 slot="kitchensink/geo-heading" class="kitchensink-heading-1">Map</h1>
```
Comment (warpech) [inline src/KitchenSink/wwwroot/KitchenSink/GeoPage.html]: Please remove the other file `MapPage.html`. It seems to be unused

File: src/KitchenSink/wwwroot/KitchenSink/NavPage.html (hunk 19 above)
Comment (PatrykSzwer) [inline src/KitchenSink/wwwroot/KitchenSink/NavPage.html]: I though my icons are cool 😞. Ok I'll remove them.

Comment (PatrykSzwer) [inline src/KitchenSink/wwwroot/KitchenSink/NavPage.html]: I left my little house for home page. I think that one could stay. 😃

**Pull request merged**